homeassistant/components/brunt/__init__.py

```diff
@@ -1 +1,68 @@
 """The brunt component."""
+import asyncio
+import logging
+
+from aiohttp.web_exceptions import HTTPError
+from brunt import BruntClientAsync
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+PLATFORMS = ["cover"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Component setup, run import config flow for each entry in config."""
+    if DOMAIN not in config:
+        return True
+    hass.data.setdefault(DOMAIN, {})
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Brunt using config flow."""
+    hass.data.setdefault(DOMAIN, {})
+    session = async_get_clientsession(hass)
+    bapi = BruntClientAsync(
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        session=session,
+    )
+    try:
+        await bapi.async_login()
+    except HTTPError as exc:
+        raise ConfigEntryAuthFailed(
+            f"Brunt could not connect with username: {entry.data[CONF_USERNAME]}."
+        ) from exc
```

Comment:
If you want to catch all scenario's nicely, perhaps you could add a exception catcher for when there is no internet connectivity. (since you do cloud pull)

```python3
except (TimeoutError, ClientError, ServerDisconnectedError) as exception:
        raise ConfigEntryNotReady("Failed to connect") from exception
    except Exception as exception:  # pylint: disable=broad-except
        _LOGGER.exception(exception)
        return False
```